src/libultra_code/cosf.c

```diff
@@ -0,0 +1,95 @@
+#include "libultra_internal.h"
+#include "guint.h"
+
+#pragma weak fcos = __cosf
+#pragma weak cosf = __cosf
+#define fcos __cosf
+
+/* coefficients for polynomial approximation of cos on +/- pi/2 */
+
+static const du P[] = {
+    { 0x3ff00000, 0x00000000 }, { 0xbfc55554, 0xbc83656d }, { 0x3f8110ed, 0x3804c2a0 },
+    { 0xbf29f6ff, 0xeea56814 }, { 0x3ec5dbdf, 0x0e314bfe },
+};
+
+static const du rpi = { 0x3fd45f30, 0x6dc9c883 };
+
+static const du pihi = { 0x400921fb, 0x50000000 };
+
+static const du pilo = { 0x3e6110b4, 0x611a6263 };
+
+static const fu zero = { 0x00000000 };
+
+/* ====================================================================
+ *
+ * FunctionName		fcos
+ *
+ * Description		computes cosine of arg
+ *
+ * ====================================================================
+ */
+
+float fcos(float x) {
+    float absx;
+    double dx, xsq, poly;
+    double dn;
+    int n;
+    double result;
+    int ix, xpt;
```

Comment:
types from `types.h` should be used here and below
```suggestion
f32 fcos(f32 x) {
    f32 absx;
    f64 dx, xsq, poly;
    f64 dn;
    s32 n;
    f64 result;
    s32 ix, xpt;
```